### PR TITLE
feat: actions updates

### LINKED ISSUE #1049: improve actions
**Please describe the feature**

As a developer, I want the GH Actions to be communicative in the PR and actionable, so that I know if something needs to be done, and what suggestions are provided.

**Acceptance Criteria**

- [x] all Actions should run on all PRs
- [x] any Actions warning/error is posted as a comment in the PR thread or reported as an Actions error
- [x] Actions run fast so that there is little lag time between opening a PR and getting feedback from Actions
- [x] The changes do not affect the deployment
- [x] The changes do not reduce the performance

**Additional context**

Some Actions were stuck on running on `develop`, but that branch was removed.

### PR DESCRIPTION
**Pull request by mihai-sysbio**
**Related issue(s) and PR(s)**  
This PR closes  #1049 by upgrading the CodeQL action (warning existed), updating the Actions so they run on the PRs against main (thus all PRs), updating the Prettier action so that it tries to fix formatting warning/errors, and finally updating the ESLint action to also output the suggestion in line with code.

**Type of change**  

- [x] Other: this is only for development and not the website itself
 
**List of changes made**  

**Screenshot of the fix**  

**Testing**  

- Instructions on how to test: have a look at the Actions output. If you have prettier and linting running when you make local commits, commit using --no-verify

**Further comments**  

There is the option to auto-commit Prettier fixes. It's relinquishing control, but improving the speed.

**Definition of Done checklist**  
- [x] My code meets the Acceptance Criteria
- [x] My code follows the [NBIS style guidelines](https://github.com/NBISweden/development-guidelines)
- [x] I have performed a self-review of my own code and commented any hard-to-understand areas
- [x] Tests and lint/format validations are passing
- [x] My changes generate no new warnings

### CODE REVIEW

**Comment by inghylt** [comment]
To try these changes out, would it make sense to push faulty changes to this branch and see if they are caught by the gh actions? And then remove the faulty commits

**Comment by mihai-sysbio** [comment]
Absolutey @inghylt, please try it out! (that's also part of the reason for those force-pushes)

File: frontend/src/components/shared/RangeFilter.vue [hunk 1]
```diff
@@ -69,7 +69,7 @@ export default {
   },
   methods: {
     inputValid(maybeNumber) {
-      if (maybeNumber === null || maybeNumber === "") {
+      if (maybeNumber === null || maybeNumber === '') {
```
Comment (mihai-sysbio) [inline frontend/src/components/shared/RangeFilter.vue:72]: Here's an example of a fix that was done automatically for me.

**Comment by MalinAhlberg** [comment]
I would have expected this commit https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1050/commits/0ea2cebc7da7a3a87952e4f6b8d97fff4c8f6480 to get an error from eslint, as it does locally.
I would also have expected this commit 420fb35 to be automatically fixed by prettier.

**Comment by mihai-sysbio** [comment]
> I would have expected this commit [0ea2ceb](https://github.com/MetabolicAtlas/MetabolicAtlas/commit/0ea2cebc7da7a3a87952e4f6b8d97fff4c8f6480) to get an error from eslint, as it does locally.

For some reason, in the [action](https://github.com/MetabolicAtlas/MetabolicAtlas/runs/7744812643?check_suite_focus=true) the file is detected as changed but not linted:
```
Run sibiraj-s/action-eslint@v2
  with:
    github-token: ***
    eslint-args: --config frontend/.eslintrc.js --resolve-plugins-relative-to frontend frontend/src/**
    extensions: .js,.vue
    annotations: true
    bin-path: frontend/node_modules/eslint/bin
Files changed.
  - frontend/src/content/about.js
Notice: No files found. Skipping.
```

Will try to remove the `.` before the extension, since that seems to be how it is defined in the [documentation](https://github.com/sibiraj-s/action-eslint#usage).

File: .github/workflows/eslint.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+          cd frontend
+          npm install --location=global yarn
+          yarn install --frozen-lockfile
+
+      - name: "/frontend: Linting"
+        uses: sibiraj-s/action-eslint@v2
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }} # Optional
+          eslint-args: "--config frontend/.eslintrc.js --resolve-plugins-relative-to frontend frontend/src/**"
+          extensions: ".js,.vue"
```
Comment (mihai-sysbio) [inline .github/workflows/eslint.yml:26]:
```suggestion
          extensions: "js,vue"
```

File: .github/workflows/eslint.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+          cd api
+          npm install --location=global yarn
+          yarn install --frozen-lockfile
+
+      - name: "/api: Linting"
+        uses: sibiraj-s/action-eslint@v2
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }} # Optional
+          eslint-args: "--config api/.eslintrc.js --resolve-plugins-relative-to api api/src/**"
+          extensions: ".js"
```
Comment (mihai-sysbio) [inline .github/workflows/eslint.yml:50]:
```suggestion
          extensions: "js"
```

File: .github/workflows/prettier.yml [hunk 4]
```diff
@@ -1,43 +1,27 @@
...
-        run: yarn install --frozen-lockfile
-
-      - name: "/api: Check formatting"
-        working-directory: ./api
-        run: yarn run check-format
-
-      - name: "/api: Linting"
-        working-directory: ./api
-        run: yarn run lint
+          prettier_options: "--check frontend/src --config frontend/.prettierrc.json --ignore-path frontend/.prettierignore "
```
Comment (mihai-sysbio) [inline .github/workflows/prettier.yml:14]:
```suggestion
          prettier_options: "--write frontend/src --config frontend/.prettierrc.json --ignore-path frontend/.prettierignore "
```

File: .github/workflows/prettier.yml [hunk 5]
```diff
@@ -1,43 +1,27 @@
...
+          # working_directory: frontend/
+          only_changed: False
+          dry: False
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: "/api: Prettify code"
+        uses: creyD/prettier_action@v4.2
+        with:
+          prettier_options: "--check api/src api/test --config api/.prettierrc.json --ignore-path api/.prettierignore"
```
Comment (mihai-sysbio) [inline .github/workflows/prettier.yml:23]:
```suggestion
          prettier_options: "--write api/src api/test --config api/.prettierrc.json --ignore-path api/.prettierignore"
```

**Comment by mihai-sysbio** [comment]
> I would also have expected this commit [420fb35](https://github.com/MetabolicAtlas/MetabolicAtlas/commit/420fb358820569f3e153108d979d0b48c7703ee7) to be automatically fixed by prettier.

The expectation is now met ☺️ see commit [6e684a4](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/1050/commits/6e684a4d166d2ca3005247a4e832f94d49ed0a63).

**Comment by mihai-sysbio** [comment]
From what I'm seeing in the linter action, it checks only the new files on push, so your new change triggered it @MalinAhlberg 🎉

**Comment by MalinAhlberg** [comment]
Nice work @mihai-sysbio ! (And sorry for the force-pushing... I have your original version backed-up locally, at least). Prettier works very well now, but eslint seems confused. I can have a closer look at it later.

**Comment by mihai-sysbio** [comment]
@MalinAhlberg, if I understand things right, the commit 170dacf that reverts the removal of `.` goes against the [documentation](https://github.com/sibiraj-s/action-eslint#usage) and the [source code of the action](https://github.com/sibiraj-s/action-eslint/blob/1e1b44c70776726531fce03718ef3af9979d58fe/src/eslint.ts#L28).

**Comment by MalinAhlberg** [comment]
> @MalinAhlberg, if I understand things right, the commit [170dacf](https://github.com/MetabolicAtlas/MetabolicAtlas/commit/170dacf2a8b235e461bd32dc0ab64ea8cb7393ff) that reverts the removal of `.` goes against the [documentation](https://github.com/sibiraj-s/action-eslint#usage) and the [source code of the action](https://github.com/sibiraj-s/action-eslint/blob/1e1b44c70776726531fce03718ef3af9979d58fe/src/eslint.ts#L28).

@mihai-sysbio yes, we're experimenting a bit to see if we can the linter to run properly, to start with. We will revert again (or somehow clean up) once we have something working.

**Comment by inghylt** [comment]
@mihai-sysbio @e0 @nanjiangshu

@MalinAhlberg and me didn't manage to fix the problems with eslint using `sibiraj-s/action-eslint@v2`, so we decided to use what worked before but keeping all the nice changes @mihai-sysbio introduced in this PR with the new file structure and changes to CodeQL and prettier. The current eslint setup displays the error as seen in the picture below, we think that can be considered good enough for now. We created #1076 for future exploration 

image

File: .github/workflows/eslint.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: "16.x"
+
+      - name: "/api: Install dependencies"
+        run: |
+          cd api
```
Comment (mihai-sysbio) [inline .github/workflows/eslint.yml:36]: For consistency with the rest of the workflow:

```suggestion
        working-directory: api
        run: |
```

**Review by mihai-sysbio — COMMENTED** [review]
The major thing missing from the previous implementation was that the workflow could fail but it would still report as passing overall.

File: .github/workflows/eslint.yml (hunk 6 above)
Comment (inghylt) [reply .github/workflows/eslint.yml]: Nice catch! Tested it and seems to work as expected :+1:

**Review by MalinAhlberg — APPROVED** [review]
Very nice changes from @mihai-sysbio, it's really nice to have prettier commit automatically!
eslint also seems to behave good now.

**Comment by nanjiangshu** [comment]
> @mihai-sysbio @e0 @nanjiangshu
> 
> @MalinAhlberg and me didn't manage to fix the problems with eslint using `sibiraj-s/action-eslint@v2`, so we decided to use what worked before but keeping all the nice changes @mihai-sysbio introduced in this PR with the new file structure and changes to CodeQL and prettier. The current eslint setup displays the error as seen in the picture below, we think that can be considered good enough for now. We created #1076 for future exploration
> 
> image

Hi @inghylt When I tested, [linting on GH](https://github.com/MetabolicAtlas/MetabolicAtlas/actions/runs/2845672609) does not report error as shown in the screenshot you attached. Is that because you have changed the linting setting to standard github action and is that expected behavior?

**Comment by MalinAhlberg** [comment]
@nanjiangshu it looks like you change the variable to the string `'katt'`, which is fine by the linter. If you change it to the variable `katt`, you should see the error.

**Comment by nanjiangshu** [comment]
> @nanjiangshu it looks like you change the variable to the string `'katt'`, which is fine by the linter. If you change it to the variable `katt`, you should see the error.

Aha, I see. It makes sense. Otherwise linting will need to compile the code 😃

**Review by nanjiangshu — APPROVED** [review]
All look good to me!

**Pull request merged**